src/App.jsx

```diff
@@ -1,36 +1,85 @@
 import { useEffect, useState } from "react";
 import RestaurantButton from "./Components/RestaurantButton/index.jsx";
 import Hero from "./Components/Hero/index.jsx";
+import MenuItems from "./Components/MenuItems/index.jsx";
+import ReturnButton from "./Components/ReturnButton/index.jsx";
 
 function App() {
-  const [restaurantNames, setRestaurantNames] = useState([]);
-
+  const [restaurantInfo, setRestaurantInfo] = useState([]);
+  const [currentId, setCurrentId] = useState(0);
+  const [currentName, setCurrentName] = useState("");
+  
+  let xlMediaColumn;
+  if (currentId) { xlMediaColumn = "xl:grid-cols-6"};
 
   useEffect(() => {
-    fetch("https://food-delivery-api.dev.io-academy.uk/restaurants")
-      .then((response) => response.json())
-      .then((data) => {
-        setRestaurantNames(data);
-      });
-  }, []);
+    if (!currentId) {
+      fetch("https://food-delivery-api.dev.io-academy.uk/restaurants")
+        .then((response) => response.json())
+        .then((data) => {
+          setRestaurantInfo(data);
+        });
+    } else {
+      fetch(
+        `https://food-delivery-api.dev.io-academy.uk/restaurants/${currentId}`
+      )
+        .then((response) => response.json())
+        .then((data) => {
+          setRestaurantInfo(data.foodItems);
+          setCurrentName(data.restaurant);
+        });
+    }
+  }, [currentId]);
+
+  function renderContent() {
+    if (!currentId) {
+      return restaurantInfo.map((restaurant) => (
+        <RestaurantButton
+        key={restaurant.id}
+        restaurantName={restaurant.name}
+        id={restaurant.id}
+        clickHandler={setCurrentId}
+        />
+      ));
+    } else {
+      return restaurantInfo.map((foodItem, index) => (
+        <MenuItems
+        key={index}
+        foodName={foodItem.foodName}
+        foodType={foodItem.foodType}
+        calories={foodItem.calories}
+        side={foodItem.sideItem}
+        price={Number(foodItem.price).toFixed(2)}
+        />
+      ));
+    }
+  }
+  
+  function renderReturnButton() {
+    return currentId ? (
+      <ReturnButton returnClickHandler={setCurrentId} />
+    ) : null;
+  }
 
+  function renderRestaurantName() {
+    if (!currentId) {
+      return <Hero currentId={currentId} heroText={"Food. Delivered."} />;
+    } else {
+      return <Hero currentId={currentId} heroText={currentName} />;
+    }
+  }
+  
   return (
     <>
-      <header className="p-4 text-center shadow-lg md:text-left">
+      <header className="p-4 text-center shadow-lg md:flex md:justify-between">
         <p>
           <span className="text-cyan-500">Food</span>Delivery
         </p>
+        {renderReturnButton()}
```

Comment:
This function could simply be replaced with a ternary inline. This would be more 'reacty', and actually clearer I think to most React devs:

```suggestion
      {currentId ? (
        <Hero currentId={currentId} heroText={currentName} />
      ) : (
        <Hero currentId={currentId} heroText={"Food. Delivered."} />
      )}
```

Reply:
done